### ISSUE TITLE
Print warning if --hidden is not set and search pattern starts with a dot

### ISSUE THREAD

**Issue by Porkepix** [sharkdp/fd#615]
I doon't know if this is a correct idea considering the double meaning the dot represent: both "any character" for its regex meaning and the fact it's an hidden file/folder.

However, searching today after for example `.gitlab-ci.yml` files in a whole tree, it didn't found anything at first, well because hidden file, so `-H` flag is needed.

I considered it could be a nice addition, if string start exactly by a dot to enable the flag, or not exclude hidden files with exact matching. This can however conflict with the "any character" meaning of the dot, so I'll let you decide whether this is a good idea or not.

**Comment by sharkdp** [comment]
Thank you for your feedback.

I like the idea in general, but I certainly don't want to enable it for a pattern like `.gitlab…` because, as you said, the `.` here represents something else (match any character). In addition, it would also match files like `foogitlab`, because there is no `^` sign in the beginning of this pattern. The way you would actually search for files being called `.gitlab-ci.yml` would be via:
```bash
fd -H '^\.gitlab-ci\.yml$'
```
Having to escape all of this by hand is not very convenient, of course. This is why we have the `--fixed-strings` option. However, I now tend to use the `-g`/`--glob` option if I want to search for exact matches:
```bash
fd -H .gitlab-ci.yml
```

So what we could do would be the following:

If the user does not pass the `--hidden`/`-H` option and
- supplies a regex pattern that *only* matches files starting with an actual dot (like `^\.xxx`) or
- supplies a glob pattern via `-g .xxx` that starts with a dot
we could print a warning that no results will be found, because search for hidden files is enabled.

We would need to take about platform-inconsistencies as well. Hidden files on Windows do mean something else.

**Comment by sharkdp** [comment]
This is now implemented:
```
▶ fd -g .gitlab-ci.yml 
[fd error]: The pattern seems to only match files with a leading dot, but hidden files are filtered by default. Consider adding -H/--hidden to search hidden files as well or adjust your search pattern.

▶ fd '^\.gitlab-ci\.yml$'   
[fd error]: The pattern seems to only match files with a leading dot, but hidden files are filtered by default. Consider adding -H/--hidden to search hidden files as well or adjust your search pattern.
```

**Comment by sharkdp** [comment]
released in [v8.2.0](https://github.com/sharkdp/fd/releases/tag/v8.2.0).
